engine/record_tx_repository.go

```diff
@@ -0,0 +1,53 @@
+package engine
+
+import (
+	"context"
+	"iter"
+	"slices"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/database"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models/bsv"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type recordTXRepository struct {
+	db *gorm.DB
+}
+
+// SaveTX saves a transaction to the database.
+func (r *recordTXRepository) SaveTX(ctx context.Context, txTable *database.Transaction) error {
```

Comment:
maybe better name for `txTable` would be `txRecord` or `txRow`

Reply:
Done 😄